parachain/pallets/omni-account/src/lib.rs

```diff
@@ -119,13 +119,25 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		/// An account store is created
-		AccountStoreCreated { who: T::AccountId },
+		AccountStoreCreated { who: T::AccountId, account_store: MemberAccounts<T> },
 		/// Some member account is added
-		AccountAdded { who: T::AccountId, member_account_hash: H256 },
+		AccountAdded {
+			who: T::AccountId,
+			member_account_hash: H256,
+			account_store: MemberAccounts<T>,
```

Comment:
Why does it need to contain all accounts ? Does it make sense to have only added account here ?
For a long run it doesn't seem to be efficient.

If we decide to keep it, maybe worker is only interested in `AccountStoreChanged` event ? 
It could be emitted in case of every account store modification (4 event types reduced to 1)

Reply:
I had a similar thought, the idea is that we don't have to manually mutate the in-memory state in the worker as well. Also, since we don't re-validate the data cached it would be safer in case the worker gets out of sync for some reason as the last event for an account would always keep it up to date. If the concern is that users could have too many accounts I think we could reduce the limit to something more realistic, we currently have the same limit as the IDGraphs (64) but for the omni-account it probably does not make much sense.

I understand the idea to reduce the events to just one, but I think having separate events also makes it easier for us to understand the user's behaviour (if we ever want to track this)

Reply:
There can be 5 event types, 4 indicating action ( ex `AccountAdded`) and one indicating sideeffect (`AccountStoreChanged`).

But I'm still more concerned about event size in this case, if there is alternative we should explore it.

Reply:
> There can be 5 event types, 4 indicating action ( ex AccountAdded) and one indicating sideeffect (AccountStoreChanged).

True, that's a good idea.

> But I'm still more concerned about event size in this case, if there is alternative we should explore it.

Yeah that's a valid concern. Let's see what else we can do, maybe having an aggressive cache re-validation  or simply request the data on demand would not be that bad? There are some tradeoff we should consider.

But I think in reality, let's say on average users have 4 or even 10 accounts, I don't think that would be too much of an issue

Reply:
Yes I'd say we are safe for now - it's a boundedVec too, we can always limit the size